stl/inc/xmemory

```diff
@@ -1112,53 +1112,58 @@ public:
     }
 
     _CONSTEXPR20_CONTAINER _Iterator_base12& operator=(const _Iterator_base12& _Right) noexcept {
-        if (_Myproxy != _Right._Myproxy) {
-            if (_Right._Myproxy) {
-                _Adopt(_Right._Myproxy->_Mycont);
-            } else { // becoming invalid, disown current parent
 #if _ITERATOR_DEBUG_LEVEL == 2
-                _Orphan_me_v2();
-#else // _ITERATOR_DEBUG_LEVEL == 2
+#ifdef __cpp_lib_constexpr_dynamic_alloc
+        if (_STD is_constant_evaluated()) {
+            _Assign_unlocked(_Right);
+        } else
+#endif // __cpp_lib_constexpr_dynamic_alloc
+        {
+            _Assign_locked(_Right);
+        }
+#else // ^^^ _ITERATOR_DEBUG_LEVEL == 2 ^^^ / vvv _ITERATOR_DEBUG_LEVEL != 2 vvv
+        if (_Myproxy != _Right._Myproxy) {
+            if (_Right._Myproxy && _Right._Myproxy->_Mycont) {
+                _Myproxy = _Right._Myproxy->_Mycont->_Myproxy;
+            } else {
                 _Myproxy = nullptr;
-#endif // _ITERATOR_DEBUG_LEVEL == 2
             }
         }
+#endif // _ITERATOR_DEBUG_LEVEL != 2
         return *this;
     }
 
 #if _ITERATOR_DEBUG_LEVEL == 2
     _CONSTEXPR20_CONTAINER ~_Iterator_base12() noexcept {
-        _Orphan_me_v2();
+#ifdef __cpp_lib_constexpr_dynamic_alloc
+        if (_STD is_constant_evaluated()) {
+            _Orphan_me_unlocked();
+        } else
+#endif // __cpp_lib_constexpr_dynamic_alloc
+        {
+            _Orphan_me_locked();
+        }

```

Comment:
Note this is a little bit of unrelated debug performance improvement, but I believe we should try to minimize the effect on debug performance